### PR TITLE
ci : 91 conventional commit semantic release

### LINKED ISSUE #91: Enable conventional commit & semantic release workflow
- Ensure conventional commit is enforced in the rule checks
- Verify windows and mac cross-compile is working and generating all needed outputs

See example in [solo](https://github.com/hashgraph/solo/blob/main/.github/workflows/flow-deploy-release-artifact.yaml) for semantic release / conventional commits

Implemented using the approach in `full-stack-testing` repo.

Tested using dry-run enabled and by changing the `.releaserc` config file to be individual branch name instead of `main`.

Commented out the actual publish steps as we did not want to publish from a private branch.

### PR DESCRIPTION
**Pull request by andrewb1269**
**Description**:

The purpose of this PR is to implement semantic releases for the `hedera-cryptography` repository.

**Related Issue(s)**:
 Fixes #91

### CODE REVIEW

File: .github/workflows/publish_release.yml [hunk 1]
```diff
@@ -45,6 +55,78 @@ jobs:
...
+              sudo apt update
+            echo "::endgroup::"
+            echo "::group::Installing GnuPG Tools"
+              sudo apt install -y gnupg2
+            echo "::endgroup::"
+          fi
+
+      - name: Import GPG key
+        id: gpg_key
+        uses: crazy-max/ghaction-import-gpg@cb9bde2e2525e640591a934b1fd28eef1dcaf5e5 # v6.2.0
```
Comment (andrewb1269) [inline .github/workflows/publish_release.yml]: Update this to the `step-security` instead.
`step-security/ghaction-import-gpg`

File: .github/workflows/publish_release.yml [hunk 2]
```diff
@@ -45,6 +55,78 @@ jobs:
...
+          git_tag_gpgsign: false
+
+      - name: Setup Node
+        uses: actions/setup-node@39370e3970a6d050c480ffad4ff0ed4d3fdee5af # v4.1.0
+        with:
+          node-version: 20.18.0
+
+      - name: Calculate Next Version
+        env:
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (andrewb1269) [inline .github/workflows/publish_release.yml]: Potentially add `NPM_TOKEN` to this repo.

File: .github/workflows/publish_release.yml [hunk 3]
```diff
@@ -45,6 +55,78 @@ jobs:
...
+      - name: Setup NPM Authentication
+        run: |
+          # create .npmrc file to publish solo to Github NPM registry
+          rm -f .npmrc || true
+          echo '//registry.npmjs.org/:_authToken=${NPM_TOKEN}' > ".npmrc"
+          echo '@hashgraph:registry=https://registry.npmjs.org/' >> ".npmrc"
+
+      - name: Publish Semantic Release
+        env:
+          commit: ${{ needs.update-readme.outputs.commit-hash }}
```
Comment (andrewb1269) [inline .github/workflows/publish_release.yml]: Remove the `commit` variable.

File: .github/workflows/publish_release.yml (hunk 1 above)
Comment (andrewb1269) [reply .github/workflows/publish_release.yml]: Commit ID: `6c8fe4d0126a59d57c21f87c9ae5dd3451fa3cca # v.6.1.0`

**Comment by andrewb1269** [comment]
## BEFORE MERGING, ENSURE THE `.RELEASERC` FILE IS CHANGED BACK TO MAINLINE INSTEAD OF PRIVATE BRANCH.

**Review by rbarker-dev — APPROVED** [review]
LGTM

**Review by jeromy-cannon — APPROVED** [review]

**Pull request merged**